### PR TITLE
Psi4 is back

### PR DESCRIPTION
**Pull request by alongd**
### Motivation or Problem
This is a replacement of the stall PR #2024, enhancing Arkane with an ESS adapter for Psi4.
The previous branch was rebased and modified.
Code originally written by @dranasinghe !

### Description of Changes
- A Psi4 adapter was added with tests
- Orca Hessian and Modes reading methods were added
- Minor additional modifications to Arkane ESS adapters

### Testing
Tests were added to all major new features.

### CODE REVIEW

**Review by kfir4444 — APPROVED** [review]
Thanks @alongd ! I'm excited to revisit Psi4!

**Comment by alongd** [comment]
Tagging @kspieks as the original reviewer, @dranasinghe as the original author, and @kfir4444 who's implementing Psi4 in ARC. Looking forward to hearing your comments.

File: arkane/commonTest.py
Comment (kspieks) [inline arkane/commonTest.py:50]: To be overly particular, can we import these in alphabetical order i.e. put `convert_imaginary_freq_to_negative_float` before all of the `get_` functions?

File: arkane/ess/factory.py
Comment (kspieks) [inline arkane/ess/factory.py:107]: Very minor: can we add a comma between Psi4 and QChem?

File: arkane/ess/psi4.py
Comment (kspieks) [inline arkane/ess/psi4.py:60]: Minor typo: should be `Psi4 log file`

File: arkane/ess/psi4.py
Comment (kspieks) [inline arkane/ess/psi4.py:68]: After checking for normal completion, can we add an elif to look for some specific errors and then print that line for user convenience? For example, it seems that keywords indicating errors include `RuntimeError`, `Fatal Error`, and/or `PSIO Error`. Can you add a unit test for the `check_for_errors` method and perhaps parse the failed output file from this discussion board ([link](http://forum.psicode.org/t/psio-error-during-adc-2/1338))? Feel free to add any failed runs from your work.

File: arkane/ess/psi4Test.py
Comment (kspieks) [inline arkane/ess/psi4Test.py:44]: Minor typo: should be Psi4

File: arkane/ess/psi4Test.py
Comment (kspieks) [inline arkane/ess/psi4Test.py:54]: Can we add a unit test for the `load_zero_point_energy` and `load_force_constant_matrix` methods? Your current output files have the necessary information so we can just parse those for the test

**Review by kspieks — CHANGES_REQUESTED** [review]
Thanks for this PR! Great to see that psi4 is back! The PR is very organized so I only left a few minor comments.

File: arkane/ess/psi4.py
Comment (alongd) [reply arkane/ess/psi4.py:68]: Excellent suggestion. I added the conditions and the online example. I don't have experience with failed Psi4 runs. @kfir4444, would you know of typical Psi4 fails and corresponding phrases in output files?

File: arkane/ess/psi4Test.py
Comment (kspieks) [inline arkane/ess/psi4Test.py:116]: Thanks for adding these tests :) Can you double check the values? Both of these unit tests are currently failing ([link](https://github.com/ReactionMechanismGenerator/RMG-Py/runs/4760479782?check_suite_focus=true#step:10:6045))

Comment (alongd) [reply arkane/ess/psi4Test.py:116]: Absolutely, these are just dummies, I pushed the tentative fixes to continue the work on a different machine :)
Sorry for not commenting, I'll tag you once the PR's ready again

Comment (kspieks) [reply arkane/ess/psi4Test.py:116]: Sorry for my misunderstanding!! Sounds good

**Comment by alongd** [comment]
@kspieks, the PR's ready, please take another look. Let me know if you have further comments, or whether I should squash.

I see that RMG-tests are unstable (also on another minor PR), I don't think it's related to this PR. I commented on the rmg slack channel.

**Comment by kspieks** [comment]
@alongd Agreed, the failing tests seem to be due to some cantera error, which is unrelated to this PR. This PR looks good so please go ahead and squash the fixup commits 😄 
I actually don't have merge access so you can feel free to merge after squashing :) Thanks for this PR!

**Comment by alongd** [comment]
@kspieks, thanks for the very helpful comments! I'm glad we brought this together to the finish line :)
I just squashed the fixups, please mark the Review as "approved" (unless there are additional comments), and I'll go ahead and merge it in

**Review by kspieks — APPROVED** [review]

**Comment by kspieks** [comment]
@alongd Thanks for squashing. I approved the PR

**Comment by mefuller** [comment]
> @alongd Agreed, the failing tests seem to be due to some cantera error, which is unrelated to this PR. This PR looks good so please go ahead and squash the fixup commits smile I actually don't have merge access so you can feel free to merge after squashing :) Thanks for this PR!

@kspieks can you expand on this and maybe open an issue for Cantera? I would certainly want to look at that and try to chase it down ASAP

**Comment by alongd** [comment]
@mefuller, see here for example:
```
-------------- start of converter log --------------
Python 3.7.11 (default, Jul 27 2021, 14:35:47) 
[GCC 7.5.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
Traceback (most recent call last):
  File "<console>", line 8, in <module>
  File "/usr/share/miniconda/envs/testing/lib/python3.7/site-packages/cantera/ctml_writer.py", line 2694, in convert
    write(outName)
  File "/usr/share/miniconda/envs/testing/lib/python3.7/site-packages/cantera/ctml_writer.py", line 377, in write
    x.write(sys.stdout)
  File "/usr/share/miniconda/envs/testing/lib/python3.7/site-packages/cantera/ctml_writer.py", line 152, in write
    filename.write(''.join(s))
BlockingIOError: [Errno 11] write could not complete without blocking
--------------- end of converter log ---------------
```
https://github.com/ReactionMechanismGenerator/RMG-tests/runs/4747926198?check_suite_focus=true

**Comment by mefuller** [comment]
0) Sorry for commenting tangentially on a merged PR, but I think this should be the end of the conversation
1) The errors have to do with `ctml_writer.py` - this is the file that Cantera uses to write `cti` files to `xml` and is deprecated as of Cantera 2.5, slated for removal in 3.0 (currently the 2.6 alpha is under development).
2) Given that, there's not much point in chasing down the bug in Cantera. The current recommendation is to use `cti2yaml.py` to write `cti` mechanisms (deprecated) to `yaml` and import those into Cantera.

**Pull request merged**
